tests/Feature/ApplicantControllerTest.php

```diff
@@ -0,0 +1,140 @@
+
+    /**
+     * Run parent setup and set global values.
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->baseUrl = 'api/v1';
+        $this->faker = \Faker\Factory::create();
+    }
+
+    public function testGetProfile(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+
+        $response = $this->actingAs($applicant->user)
+            ->getJson("$this->baseUrl/applicant/$applicant->id/profile");
+        $response->assertOk();
+        $response->assertJsonFragment([
+            'citizenship_declaration_id' => $applicant->citizenship_declaration_id,
+            'veteran_status_id' => $applicant->veteran_status_id,
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ]);
+    }
+
+    public function testUpdateProfile(): void
+    {
+        $applicant = factory(Applicant::class)->create();
+        $applicant->applicant_classifications()->delete();
+
+        // Ensure that submitting applicant classifications creates new elements.
+        $newApplicantClassification = [
+            'id' => 12345,
+            'applicant_id' => $applicant->id,
+            'classification_id' => Classification::inRandomOrder()->first()->id,
+            'level' => 5,
+            'order' => 1,
+        ];
+        $updateData = [
+            'citizenship_declaration_id' => CitizenshipDeclaration::inRandomOrder()->first()->id,
+            'veteran_status_id' => VeteranStatus::inRandomOrder()->first()->id,
+            'applicant_classifications' => [
+                0 => $newApplicantClassification
+            ],
+        ];
+        $response = $this->actingAs($applicant->user)
+            ->putJson("$this->baseUrl/applicant/$applicant->id/profile", $updateData);
+        $response->assertOk();
+
+        $expectedData = [
+            'citizenship_declaration_id' => $updateData['citizenship_declaration_id'],
+            'veteran_status_id' => $updateData['veteran_status_id'],
+            'applicant_classifications' => $applicant->applicant_classifications->toArray(),
+        ];
+        $response->assertJsonFragment($expectedData);
+        $this->assertDatabaseHas(
+            'applicant_classifications',
+            $applicant->applicant_classifications->first()->toArray()
+        );
+
+        // Ensure that submitting applicant classifications deletes elements that were not resubmitted
+        // and creates new elements/updates old elements.
+        $oldApplicantClassification = $applicant->applicant_classifications->first()->toArray();
+        $updateData = [
+            'citizenship_declaration_id' => CitizenshipDeclaration::inRandomOrder()->first()->id,
+            'veteran_status_id' => VeteranStatus::inRandomOrder()->first()->id,
+            'applicant_classifications' => [
+                0 => $oldApplicantClassification,
+                1 => $newApplicantClassification,
```

Comment:
In this case, the two classifications will be identical. Couldn't that be a problem?

Reply:
I guess, maybe not, right now? I wonder if we _should_ add some validation or database constraint stopping an applicant from adding the same classification-level combination multiple times.

Reply:
Oh ya, me and matt talked about this problem. Yes I agree, I'm thinking we could add some validation on both the back and front end.

Reply:
Turns adding the classification-level combo attributes to the `ApplicantClassification::firstOrNew` method, will filter out any duplicates.